src/cheri/xycheriot.adoc

```diff
@@ -0,0 +1,678 @@
+a bit-vector wherein each bit corresponds to a capability-sized memory granule.
+(That is, with each capability-sized and -aligned region of the primal memory.)
+Such memory block(s) are said to be _revokable_.
+These revocation bitmaps are also exposed (as memory) to software.
+
+When `{load_cap_name_lc}` transfers a capability with set {ctag} from memory
+into a hart's register file,
+it checks whether the _base_ (lower bound) of that capability is within
+a revokable memory block and, if so,
+fetches the corresponding bit in the associated revocation bitmap.
+If that bit is set, then the {ctag} stored in the hart's register file
+as part of this transfer is cleared.
+
+Software dynamic memory allocators can set bits in a revocation bitmap
+to ensure that additional copies of capabilities to particular objects
+cannot be constructed (into register files and, so, into memory, too).
+This facilitates particularly straightforward global revocation of pointers to
+freed heap objects.
+
+[NOTE]
+=====
+The location, size, and number of revokable memory blocks,
+as well as the mapping function from primal memory address to revocation bitmap
+address and bit index, are all platform-defined.
+=====
+
+[NOTE]
+=====
+See §7.8 / "Temporal safety" of {cheriot_v1_isa} for further details.
+=====
+
+=== Capability Encoding
+
+==== `mycfg` Value
+
+Harts implementing {cheriot_ext_name} must report
+a constant {cheriot_ycfg_value} in `mycfg`.
+(See <<uycfg>> for the details of this CSR.)
+
+[NOTE]
+=====
+TODO: That value is, of course, not finalized.
+=====
+
+==== Bounds Encoding
+
+CHERIoT's bounds encodings use 9-bit top and bottom mantissa fields
+and a compressed 4-bit exponent value.
+The resulting encoding significantly differs from the bounds
+of the <<app_cap_description>> encoding; CHERIoT's encoding...
+
+- has byte alignment only up to length 511.
+
+- does not guarantee a representable region beyond the authorized base and top
+  (except for "one past the end", where the address is equal to the limit).
++
+This requires using a non-default
+<<section_xycheriot_auipc_shift,AUIPC shift>> value.
+
+[NOTE]
+=====
+TODO: As in §7.13.3 / "Capability bounds" in {cheriot_v1_isa}
+=====
+
+==== Permissions
+
+{cheriot_ext_name} defines _one_ <<sec_cap_sdp,software defined permission>>,
+which we call `U0`.
+That is, its `SDPLEN` value is `1`.
+
+There are 11 other architectural permission bits in {cheriot_ext_name},
+all of which are from its constituent components:
+
+* <<r_perm>>, <<w_perm>>, <<c_perm>>, <<lm_perm>>, <<x_perm>>, and <<asr_perm>>
+  from {cheri_base_ext_name};
+
+* <<zylevels1_lg_perm>>, <<zylevels1_sl_perm>>, and <<zylevels1_gl_perm>>
+  from <<section_zylevels1>>; and
+
+* <<zyseal_se_perm>> and <<zyseal_us_perm>> from <<section_zyseal>>.
+
+The aggregate permissions bit field (<<acperm_bit_field>>) used by <<GCPERM>>
+and <<CLRPERM>> for {cheriot_ext_name} is thus:
+include::./img/xycheriot-acperm_bit_field.edn[]
```

Comment:
Do we have to specify this encoding?

Reply:
Ah, no, sorry, this was originally in a [NOTE] and I moved it out to the main text without indicating that it is purely a consequence of other parts of the spec.